pkg/function/create_csi_snapshot_static.go

```diff
@@ -0,0 +1,164 @@
+	kanister "github.com/kanisterio/kanister/pkg"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/kube/snapshot"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+func init() {
+	_ = kanister.Register(&createCSISnapshotStaticFunc{})
+}
+
+var (
+	_ kanister.Func = (*createCSISnapshotStaticFunc)(nil)
+)
+
+const (
+	// CreateCSISnapshotStaticFuncName gives the name of the function
+	CreateCSISnapshotStaticFuncName = "CreateCSISnapshotStatic"
+
+	// CreateCSISnapshotStaticNameArg provides name of the new VolumeSnapshot
+	CreateCSISnapshotStaticNameArg = "name"
+
+	// CreateCSISnapshotStaticNamespaceArg specifies the namespace of the new VolumeSnapshot
+	CreateCSISnapshotStaticNamespaceArg = "namespace"
+
+	// CreateCSISnapshotStaticDriverArg specifies the CSI driver used in the new VolumeSnapshotContent
+	CreateCSISnapshotStaticDriverArg = "driver"
+
+	// CreateCSISnapshotStaticSnapshotHandleArg specifies the UID of the backend storage snapshot used in the new VolumeSnapshotContent
+	CreateCSISnapshotStaticSnapshotHandleArg = "snapshotHandle"
+
+	// CreateCSISnapshotStaticSnapshotClassArg specifies the name of the VolumeSnapshotClass
+	CreateCSISnapshotStaticSnapshotClassArg = "snapshotClass"
+
+	// CreateCSISnapshotStaticRestoreSizeArg gives the storage size required for PV/PVC restoration
+	CreateCSISnapshotStaticRestoreSizeArg = "restoreSize"
+
+	// CreateCSISnapshotStaticSnapshotContentNameArg provides the name of dynamically provisioned VolumeSnapshotContent
+	CreateCSISnapshotStaticSnapshotContentNameArg = "snapshotContent"
+)
+
+type createCSISnapshotStaticFunc struct{}
+
+func (*createCSISnapshotStaticFunc) Name() string {
+	return CreateCSISnapshotStaticFuncName
+}
+
+func (*createCSISnapshotStaticFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+	var (
+		name, namespace                       string
+		driver, snapshotHandle, snapshotClass string
+	)
+
+	if err := Arg(args, CreateCSISnapshotStaticNameArg, &name); err != nil {
+		return nil, err
+	}
+
+	if err := Arg(args, CreateCSISnapshotStaticNamespaceArg, &namespace); err != nil {
+		return nil, err
+	}
+
+	if err := Arg(args, CreateCSISnapshotStaticDriverArg, &driver); err != nil {
+		return nil, err
+	}
+
+	if err := Arg(args, CreateCSISnapshotStaticSnapshotHandleArg, &snapshotHandle); err != nil {
+		return nil, err
+	}
+
+	if err := Arg(args, CreateCSISnapshotStaticSnapshotClassArg, &snapshotClass); err != nil {
+		return nil, err
+	}
+
+	kubeCli, err := kube.NewClient()
+	if err != nil {
+		return nil, err
+	}
+
+	dynCli, err := kube.NewDynamicClient()
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotter, err := snapshot.NewSnapshotter(kubeCli, dynCli)
+	if err != nil {
+		if errors.Is(context.DeadlineExceeded, err) {
+			timeoutMsg := "SnapshotContent not provisioned within given timeout. Please check if CSI driver is installed correctly and supports VolumeSnapshot feature"
```

Comment:
are we actually provisioning the snapshotcontent here? If not the error message seems to be confusing.